### PR TITLE
fix(build): add volar types export

### LINKED ISSUE #839: Error TS2688: Cannot find type definition file for '@oruga-ui/oruga-next/volar'.
### Overview of the problem

**Oruga** version: [X.X.X]
**Vuejs** version: [X.X.X]
**OS/Browser**:

### Description

When I add the type definition for volar in the tsconfig.ts like in the documentation https://oruga-ui.com/documentation/#volar-support

> If you are using Volar, you can specify global component types by configuring compilerOptions.types in tsconfig.json.
> 
> ```
> js
> // tsconfig.json
> {
>   "compilerOptions": {
>     // ...
>     "types": ["@oruga-ui/oruga-next/volar"]
>   }
> }
> ```

I have this error : Cannot find type definition file for '@oruga-ui/oruga-next/volar'.

### Steps to reproduce

1. Go to https://stackblitz.com/edit/vitejs-vite-dnqnu1?file=tsconfig.json 
2. Type in the terminal : npm run type-check
3. See the error 

```
❯ npm run type-check

> vite-vue-typescript-starter@0.0.0 type-check
> vue-tsc

error TS2688: Cannot find type definition file for '@oruga-ui/oruga-next/volar'.
  The file is in the program because:
    Entry point of type library '@oruga-ui/oruga-next/volar' specified in compilerOptions

  tsconfig.json:22:15
    22     "types": ["@oruga-ui/oruga-next/volar"]
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    File is entry point of type library specified here.

Found 1 error.
```

### Expected behavior

File `volar.d.ts` should be found

### Solution

If add the file in the export section of the `oruga-ui/oruga-ext/package.json` it works.

```diff
  "exports": {
    ".": {
      "types": "./dist/types/index.d.ts",
      "import": "./dist/esm/index.mjs",
      "require": "./dist/cjs/index.js"
    },
+    "./volar": {
+      "types": "./volar.d.ts"
+    },
    "./*": "./*"
  },
```

### PR DESCRIPTION
**Pull request by mlmoravek**
Fixes #839 

## Proposed Changes

- add type to package.json export 
- update docs

### CODE REVIEW

**Comment by mlmoravek** [comment]
@messenjer could you give me assign if this is the right implementation?

**Comment by messenjer** [comment]
Hello,

It's not working with 

`"types": ["@oruga-ui/oruga-next"]`

You need to change it to:  

`"types": ["@oruga-ui/oruga-next/types"]`

If I do it, and I use it with `vue-tsc`, and I expressly add an error variant boolean, it's working.

```
> vue-tsc --build --force

src/App.vue:8:52 - error TS2322: Type 'boolean' is not assignable to type 'string'.

8     <o-button label="Launch loading" size="medium" :variant="true" />
                                                     ~~~~~~~~

  node_modules/@oruga-ui/oruga-next/dist/types/components/button/Button.vue.d.ts:352:5
    352     variant: string;
            ~~~~~~~
    The expected type comes from property 'variant' which is declared here on type 'Partial<{ label: string; size: string; override: boolean; iconPack: string; variant: string; rootClass: ComponentClass; labelClass: ComponentClass; expanded: boolean; ... 22 more ...; variantClass: ComponentClass; }> & Omit<...> & Record<...>'

Found 1 error.
```

For me the best solution should be, in tsconfig

`"types": ["@oruga-ui/oruga-next/volar"]`

And in package.json

```
    "./*": "./*",
    "./volar": {
      "types": "./volar.d.ts"
    }

```

File: packages/docs/documentation/index.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ If you are using Volar, you can specify global component types by configuring `c
 {
   "compilerOptions": {
     // ...
-    "types": ["@oruga-ui/oruga-next/volar"]
+    "types": ["@oruga-ui/oruga-next"]
```
Comment (messenjer) [inline packages/docs/documentation/index.md:125]:
```suggestion
    "types": ["@oruga-ui/oruga-next/volar"]
```

File: packages/oruga/package.json [hunk 2]
```diff
@@ -16,7 +16,10 @@
       "import": "./dist/esm/index.mjs",
       "require": "./dist/cjs/index.js"
     },
-    "./*": "./*"
+    "./*": "./*",
+    "./types": {
```
Comment (messenjer) [inline packages/oruga/package.json:20]:
```suggestion
    "./volar": {
```

**Review by messenjer — COMMENTED** [review]
Here is my suggestion

**Pull request merged**